packages/webapp/src/containers/Animals/Inventory/useFilteredInventory.ts

```diff
@@ -22,7 +22,19 @@ import { animalMatchesFilter } from '../../Filter/Animals/utils';
 import { animalsFilterSelector } from '../../filterSlice';
 import { isInactive } from '../../Filter/utils';
 
-export const useFilteredInventory = (inventory: AnimalInventory[]) => {
+export const useFilteredInventory = (
+  inventory: AnimalInventory[],
+  showOnlySelected: boolean,
+  selectedInventoryIds: string[],
+) => {
+  if (showOnlySelected) {
+    return useMemo(() => {
```

Comment:
Doesn't this break the rules of hooks since you're calling a hook within a conditional? https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level

I think we could just add a condition to `animalOrBatchMatches` like so

```js
const animalOrBatchMatches =
        showOnlySelected && selectedInventoryIds.includes(entity.id) ||
        isInactive(animalsOrBatchesFilter) ||
        (entity.batch
          ? animalsOrBatchesFilter[AnimalOrBatchKeys.BATCH]?.active
          : animalsOrBatchesFilter[AnimalOrBatchKeys.ANIMAL]?.active);
```

Reply:
Wow great point about breaking the hook rules that was silly of me. I dont know why I didn't get a warning either. Will do that!

Reply:
Tried it the suggested way, but the other matches returned extra results which was not desired. 

Because selectedIds is likely to change more frequently I just moved it out of the conditional but kept its own separate useMemo.